lib/src/arithmetic/fixed_point/fixed_point_logic.dart

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// fixed_point_logic.dart
+// Implementation of Fixed Point objects
+//
+// 2024 September 25
+// Author: Soner Yaldiz <soner.yaldiz@intel.com>
+
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/src/exceptions.dart';
+
+/// Fixed point logic representation
+class FixedPoint extends Logic {
+  /// [integerWidth] is the width of bits reserved for integer part
+  final int integerWidth;
+
+  /// [fractionWidth] is the width of bits reserved for fractional part
+  final int fractionWidth;
+
+  static int _fixedPointWidth(int a, int b) => 1 + a + b;
```

Comment:
what about unsigned fixed point?

Reply:
This type only supports signed number in two's complement. Need to revise to support unsigned fixed-point numbers.

Reply:
if it's two's complement, then isn't it implicitly signed for any width anyways? i.e. there's no "sign" bit?